Terminal.Gui/Views/CollectionNavigation/ICollectionNavigator.cs

```diff
@@ -0,0 +1,49 @@
+﻿#nullable enable
+
+namespace Terminal.Gui;
+
+/// <summary>
+///     Navigates a collection of items using keystrokes. The keystrokes are used to build a search string. The
+///     <see cref="SearchString"/> is used to find the next item in the collection that matches the search string when
+///     <see cref="GetNextMatchingItem(int, char)"/> is called.
+///     <para>
+///         If the user types keystrokes that can't be found in the collection, the search string is cleared and the next
+///         item is found that starts with the last keystroke.
+///     </para>
+///     <para>If the user pauses keystrokes for a short time (see <see cref="TypingDelay"/>), the search string is cleared.</para>
+/// </summary>
+public interface ICollectionNavigator
```

Comment:
Given you have a public base class hierarchy what is the added value of those interfaces?

Reply:
The interface allows for the consuming views (`ListView` etc) to reference an interface rather than a concrete abstract base.

I prefer this as preference even though it means more classes - in general I just love interfaces for their flexibility later on.

So user can:

- Create custom implementation of matcher only 
- Create custom implementation of entire collection navigator 

This ensures maximum flexibility - user is not tied to the abstract base if they don't want to.